azure/client.go

```diff
@@ -47,8 +54,13 @@ func NewClientFromPublishSettingsFile(publishSettingsFilePath string) (Client, e
 
 //NewClientFromPublishSettingsFile creates a new azure.Client and imports the publish
 //settings from the specified file path.
-func NewClientFromPublishSettings(subscriptionId string, managementCertificatePath string) (Client, error) {
-	client := Client{}
+func NewClientFromPublishSettings(subscriptionId string, managementCertificatePath string, serviceManagementURL ...string) (Client, error) {
+	var client Client
+	if len(serviceManagementURL) > 0 {
```

Comment:
you just duplicated logic in 2 places. when you do that it means probably you should have it only in one place and these methods should call it.